reconcile/vpc_peerings_validator.py

```diff
@@ -41,13 +42,59 @@ def validate_no_internal_to_public_peerings(
                 continue
 
             valid = False
+            pair = {cluster.name, peer.name}
+            if pair in found_pairs:
+                continue
+            found_pairs.append(pair)
             logging.error(
                 f"found internal to public vpc peering: {cluster.name} <-> {peer.name}"
             )
 
     return valid
 
 
+def validate_no_public_to_public_peerings(
+    query_data: VpcPeeringsValidatorQueryData,
+) -> bool:
+    """Iterate over VPC peerings of public clusters and validate the peer is not public."""
+    valid = True
+    found_pairs: list[set[str]] = []
+    for cluster in query_data.clusters or []:
+        if (
+            cluster.internal
+            or (cluster.spec and cluster.spec.private)
+            or not cluster.peering
+        ):
+            continue
+        for connection in cluster.peering.connections or []:
+            if connection.provider not in [
+                "cluster-vpc-accepter",
+                "cluster-vpc-requester",
+            ]:
+                continue
+            connection = cast(
+                Union[
+                    ClusterPeeringConnectionClusterAccepterV1,
+                    ClusterPeeringConnectionClusterRequesterV1,
+                ],
+                connection,
+            )
+            peer = connection.cluster
+            if peer.internal or (peer.spec and peer.spec.private):
+                continue
+
+            valid = False
+            pair = {cluster.name, peer.name}
```

Comment:
This is a nit, but wanted to make sure I wasn't missing why a `set` is being used here? It doesn't seem that these values would ever be the same, or that we'd wish to dedupe `cluster.name` and `peer.name`? I was expecting something more like:

`found_pairs: set[tuple[str, str]] = set()`

The performance implications with how many clusters we have are negligible, but I want to make sure that I'm not missing some specific intent here that's not obvious to me.

Reply:
a set is used because:
```
>>> a = {'cluster1', 'cluster2'}
>>> b = {'cluster2', 'cluster1'}
>>> a == b
True
>>> a = ('cluster1', 'cluster2')
>>> b = ('cluster2', 'cluster1')
>>> a == b
False
```

Reply:
That makes sense. I was indeed missing `cluster-vpc-requester` vs. `cluster-vpc-accepter` which is what leads to the duplication. Thanks!